src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/WidgetCodeEditor.java

```diff
@@ -0,0 +1,100 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.ui.stepsdesign;
+
+import io.cdap.cdap.ui.types.NodeInfo;
+import io.cdap.cdap.ui.utils.Commands;
+import io.cdap.cdap.ui.utils.Helper;
+import io.cdap.e2e.utils.ElementHelper;
+import io.cdap.e2e.utils.SeleniumDriver;
+import io.cucumber.java.en.Then;
+import org.apache.commons.lang.StringUtils;
+import org.junit.Assert;
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebElement;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class WidgetCodeEditor {
+    private NodeInfo jSNode = new NodeInfo("JavaScript", "transform", "0");
+    private static final String defaultJsEditorVal = "/**\n" +
+            " * @summary Transforms the provided input record into zero or more output records or errors.\n" +
+            "\n" +
+            " * Input records are available in JavaScript code as JSON objects. \n" +
+            "\n" +
+            " * @param input an object that contains the input record as a JSON.   " +
+            "e.g. to access a field called 'total' from the input record, use input.total.\n" +
+            " * @param emitter an object that can be used to emit zero or more records " +
+            "(using the emitter.emit() method) or errors (using the emitter.emitError() method) \n" +
+            " * @param context an object that provides access to:\n" +
+            " *            1. CDAP Metrics - context.getMetrics().count('output', 1);\n" +
+            " *            2. CDAP Logs - context.getLogger().debug('Received a record');\n" +
+            " *            3. Lookups - context.getLookup('blacklist').lookup(input.id); or\n" +
+            " *            4. Runtime Arguments - context.getArguments().get('priceThreshold') \n" +
+            " */ \n" +
+            "function transform(input, emitter, context) {\n" +
+            "  emitter.emit(input);\n" +
+            "}";
+
+    @Then("Toggle Transform panel")
+    public void toggleTransformPanel() {
+        Commands.toggleTransformPanel();
+    }
+
+    @Then("Add JS node to canvas")
+    public void addJSNodeToCanvas() {
+        Commands.addNodeToCanvas(jSNode);
+    }
+
+    @Then("Open JS node properties")
+    public void openJSNodeProperties() {
+        Commands.openNodeProperty(jSNode);
+    }
+    @Then("Get JS editor value and compare with default JS editor value")
+    public void verifyDefaultJSEditorValue() {
+        WebElement jsEditorElement = Helper.locateElementByCssSelector("div[class*='ace-editor-ref']");
+        WebElement jsEditorContentElement = jsEditorElement.findElement(
+                By.cssSelector("div[class*='ace_text-layer']"));
+        String editorValue = ElementHelper.getElementText(jsEditorContentElement);
+        Assert.assertEquals(StringUtils.normalizeSpace(editorValue),
```

Comment:
is the `StringUtils.normalizeSpace` needed? doesn't `editorValue.equals(defaultJsEditorVal)` work?

Reply:
yes, equals(..) is not working here, so had to go for `StringUtils.normalizeSpace` .